Common/cpp/Tools/JSScheduler.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <ReactCommon/CallInvoker.h>
+
+#include <memory>
+
+namespace reanimated {
+
+class JSScheduler {
+ public:
+  explicit JSScheduler(
```

Comment:
Are we sure we can forsake the setter of jsCallInvoker in favor of a constructor?

Reply:
Yes, we can. Previously, we'd call `setJSCallInvoker` here (separately for Android and iOS):

https://github.com/software-mansion/react-native-reanimated/blob/80dfdf1306865a276498c0a44ee63b75f5e48978/android/src/main/cpp/NativeProxy.cpp#L87C1-L88

but after these changes we can simply call `std::make_shared<JSScheduler>(jsInvoker)` in NativeReanimatedModule.cpp.